### PR TITLE
Pretty much all the rest of the docs

### PR DESCRIPTION
**Pull request by james-swash**

### CODE REVIEW

File: src/admin/bots.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Bots
+
+A Bot represents a digital worker that will perform the activities that you have set up in your automation projects. Bots can interepret the instructions and actions that have been placed in an activity and run through them in the order defined within the activity.
```
Comment (jagribble) [inline src/admin/bots.md:3]: Don't need instructions.

File: src/common/sharing.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- All - The user can perform any action with the exception of being able to change ownership. With this permission you can also remove permissions from other users.
+- Write - Gives the ability to edit entities.
+- Publish: Lets the user publish entities such as activities, workflows, and apps.
+- Delete: Lets the user delete entities.
+- Execute: Allows the user to execute jobs or activities, e.g manually running a workflow or triggering a listener.
+- Read: The lowest permission; the user can only observe the relevant data.
+
+**Cascading Permissions**
+
+The addition or removal of permissions will cascade downward from entity to entity (assuming that the user keeps the default check box ticked). For example, if you have an application which relies on data from multiple data stores to function, instead of manually sharing the application and each datastore with another user, cascading permissions will apply the same level of permissions across all entities. The purpose of cascading permissions is to ensure consistency in each aspect of your work. You will always be warned when permissions are going to be changed via cascade.
```
Comment (jagribble) [inline src/common/sharing.md:22]: Will it cascade removal?

File: src/common/versioning.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+If your workflow contains activities which are set to their draft versions, then attempting to publish that workflow will publish those activities. You will be warned that this will happen.
+
+**Apps:**
+
+App versions similarly adhere to the three tiered model as described above.
+
+An entire app can be versioned and as such you can choose to deploy the individual app states should you wish to.
+
+App pages can also have versions and work in much the same way. If you publish an app as a whole, all app pages will be published. You will be warned that this will happen.
```
Comment (jagribble) [inline src/common/versioning.md:35]: They are enforced to have versions. And there is no way to publish just a page (as designed). This sound like you could version it as well as the app

File: src/projects/apps/designer/ipl.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+IPL also has an Exclusive node, similar to the :docs-link[one found]{id="projects/automation/workflows/exclusive} in automation Workflows, which allows you to branch your IPL flow off based on conditions.
+
+However, unlike automation workflows, IPL flows can branch off in parallel to allow for parallel execution, you can achieve this by just dragging multiple paths from one node in the IPL flow to two or more others. Do be aware though that once you do this, you cannot join several paths back into one.
+
+Each action in IPL can return a result, these results can then be used in the IPL flow but cannot be used outside of the IPL flow you are currently designing. If you wish for a value to be available outside of the IPL flow then you should use In Page Values.
+
+### In Page Values (IPV)
+
+In Page Values are like variables for your App, they are stored in the memory of your browser tab that you have the App open in and can have different lifetimes based on what you select. These In Page Values are available both within IPL actions as inputs but they are also available as a datasource for any App Component property that allows you to select a datachip.
```
Comment (jagribble) [inline src/projects/apps/designer/ipl.md:29]: Are stored in the indexDB of the browser. Browser tab makes it sound like it won't share across tabs

File: src/projects/automation/activities/inputs.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Inputs
+
+By adding inputs to an activity, you are telling the :docs-link[activity]{id="projects/automation/activities/activity"} to expect to be started with some variables. Depending on the data type you select, you may also be able to give an input a default value. 
+
+When you define an input, it will appear in the :docs-link[variables]{id="projects/automation/activities/variables"} tab and can be used as a variable within that activity. Like variables, an input can be described using the same $$ notation. To learn more about variables, visit [link]
```
Comment (jagribble) [inline src/projects/automation/activities/inputs.md:5]: Missing link

File: src/projects/automation/automation.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Automation
+
+
```
Comment (jagribble) [inline src/projects/automation/automation.md:3]: Weird spacing. Should keep it consistent

File: src/resources/cms/cms.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# Content Management System (CMS)
+
+CMS, or content management system, is a different way of storing data which is specifically tailored to being displayed in Toca apps.
+
+CMS' can be found primarily via the resources tab of your Toca platform but can also be accessed via your app once they have been linked.
+
+**Files** - Within your CMS you will find a dedicated file storage area. This is where you will select files from when you add content to your CMS later. Once uploaded, files can be previewed by clicking on them.
```
Comment (jagribble) [inline src/resources/cms/cms.md:7]: You can also use files via data chips inside of the app

File: src/resources/cms/types.md [hunk 8]
```diff
@@ -0,0 +1 @@
+# CMS Types
```
Comment (jagribble) [inline src/resources/cms/types.md:1]: Do you need this as you describe it in the CMS docs

File: src/projects/apps/designer/ipl.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+Behind the scenes, these In Page Values are stored in [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB), an in-browser database that can actually be viewed by opening the browser dev tools, heading to the Application tab and locating the IndexedDB section. There will be a database under this section labelled IndexDB and this is the one which IPV's from Toca Apps are stored.
+
+When you set each In Page Value, you get a choice about how long you would like the IPV value to last and there are three different options:
+- Persistent - This will live in your browser forever or until you clear your browser cache, this will likely outlive the browser tab that contains the App but this fact should not be relied on in your design
+- Session - The value will live until the user logs out or until they close the browser tab, whichever of those two comes first
```
Comment (chrismilway) [inline src/projects/apps/designer/ipl.md:35]: The user logging out doesn't clear the session - it's strictly the browser session

File: src/projects/automation/workflows/workflow.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
+# Workflow Overview
+
+A Workflow comprises an orchestrated and repeatable pattern of activities. In Toca, the Workflow is the highest level of automation. It is the final stage where you put all your activities together to represent your business process. The workflow may be considered a view or representation of the real work.
+
+**Nodes**
+Nodes can be dragged onto your workflow canvas.
+- Start - Denotes the start point of your workflow. Can only be used once.
+- Exclusive - Defines a decision point in the workflow
+- Trigger - Allows you to start another workflow
+- label - A simple text label with customisable font options.
```
Comment (chrismilway) [inline src/projects/automation/workflows/workflow.md:10]: Any reason this is lowercase?

File: src/projects/automation/workflows/workflow.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+- Stop - Denotes the point at which the automation should stop. A workflow can have multiple stop nodes.
+
+**Error handling**
```
Comment (chrismilway) [inline src/projects/automation/workflows/workflow.md:13]: This section seems more geared towards activities.

File: src/projects/automation/workflows/workflow.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+By adding inputs to an activity, you are telling the activity to expect to be started with some variables. More on [inputs]
+
+To output a value is to say that you want to use that value somewhere outside of that activity. More on [outputs]
+
+These concepts are very important when working with workflows as they are one of the easiest ways to use contextual data across multiple activities. Simply output a datachip from one activity and then configure one or more activities further down the line to expect it as an input.
+
+An example would be: If you were to create an App User in 'Activity A', using the 'create app user' action, you could then output the subsequent 'AppUserId'. Then, further along the workflow, you could tell 'Activity B' to expect 'AppUserId' as an input. 'Activity B' could then, for example, manage the permissions of that App User, or add them to a certain group based on other contextual data.
+
+**Exclusives / Making a decision point** -
```
Comment (chrismilway) [inline src/projects/automation/workflows/workflow.md:32]: This should probably link to the full page about exclusives

File: src/common/sharing.md (hunk 2 above)
Comment (james-swash) [reply src/common/sharing.md:22]: It will indeed if you check the cascade options

File: src/projects/automation/workflows/workflow.md (hunk 11 above)
Comment (james-swash) [reply src/projects/automation/workflows/workflow.md:13]: I know, but think it is worth talking about here

Comment (chrismilway) [reply src/projects/automation/workflows/workflow.md:13]: in general terms yes, but you're describing minutiae that cannot be configured via the workflow UI.

Comment (james-swash) [reply src/projects/automation/workflows/workflow.md:13]: Okay, I concede, have moved this to the "Activity" page

File: src/resources/cms/types.md (hunk 8 above)
Comment (james-swash) [reply src/resources/cms/types.md:1]: Think I will move it from CMS Docs and place it here as I think Types will go through a bit more embellishment and maybe some diagrams in the next iteration

**Review by jagribble — APPROVED** [review]

**Pull request merged**